File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l=0 , r= numbers.length-1;
        int sum =0;
        while(l<r){
           sum = numbers[l] + numbers[r];
           if(sum == target){
            return new int[]{l+1,r+1};
           }
           if(sum > target){
             r-=1;
           }else{
             l+=1;
           }
        }

        return new int[0];
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_map = {}

        for idx,num in enumerate(nums):
            rem = target - num
            if rem in num_map:
                return [num_map.get(rem),idx]
            else:
                num_map[num] = idx

        return [-1,-1]